### PR TITLE
Fix unstable spamassassin tests

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/SpamAssassinContract.java [hunk 1]
```diff
@@ -652,22 +652,16 @@ default String setMessageCreateToMultipleRecipients(AccessToken accessToken) {
...
-            .then()
-                .statusCode(200)
-                .body(NAME, equalTo("messageList"))
-                .body(ARGUMENTS + ".messageIds", hasSize(expectedNumberOfMessages));
-            return true;
-
-        } catch (AssertionError e) {
-            return false;
-        }
+    default void assertAreMessagesFoundInMailbox(AccessToken accessToken, String mailboxId, int expectedNumberOfMessages) {
```
Comment (rouazana) [inline server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/SpamAssassinContract.java]: assertThat? assertThereAre?

File: third-party/spamassassin/src/main/java/org/apache/james/spamassassin/SpamAssassinInvoker.java [hunk 2]
```diff
@@ -237,8 +244,9 @@ private boolean reportMessageAs(InputStream message, Username username, MessageC
             out.flush();
             socket.shutdownOutput();
 
-            return in.lines()
-                .anyMatch(this::hasBeenSet);
+            boolean hasBeenSet = in.lines().anyMatch(this::hasBeenSet);
+            LOGGER.debug("Reported mail as {} succeeded", messageClass);
```
Comment (rouazana) [inline third-party/spamassassin/src/main/java/org/apache/james/spamassassin/SpamAssassinInvoker.java]: you don't show hasBeenSet in the debug statement?

File: server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/SpamAssassinContract.java [hunk 3]
```diff
@@ -672,4 +685,8 @@ default String getSpamId(AccessToken accessToken) {
     default String getTrashId(AccessToken accessToken) {
         return getMailboxId(accessToken, Role.TRASH);
     }
+
+    default void assertEveryListenersGotCalled(GuiceJamesServer jamesServer) {
+        assertThat(jamesServer.getProbe(SpoolerProbe.class).processingFinished()).describedAs("waiting that every listeners get called").isTrue();
```
Comment (rouazana) [inline server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/SpamAssassinContract.java]: cr/lf

**Comment by Arsnael** [comment]
tests are :green_apple: , deployment tests failed
```
[ERROR]   JamesDeploymentValidationTest>DeploymentValidation.validateDeployment:81 » InvalidServerResponse
```
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by trantienduchn** [comment]
```
[7f794a8c431a9f3b7aa98dd17a35d85ce24790f4] Location: /org/apache/james/imap/scripts/ValidateDeployment.test:499
[7f794a8c431a9f3b7aa98dd17a35d85ce24790f4] LastClientMsg: c SEARCH FLAGGED
[7f794a8c431a9f3b7aa98dd17a35d85ce24790f4] Expected: '\* SEARCH 1 2 3 4'
[7f794a8c431a9f3b7aa98dd17a35d85ce24790f4] Actual   : '* SEARCH'
```
`wait 10000` can not get rid of this error. test this please

**Comment by mbaechler** [comment]
failed due to github ratelimit

test this please

**Comment by Arsnael** [comment]
@mbaechler : I tried to merge it but there is rebase issues with the squashing. Can you squash your commits first? Thanks !

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**